frontend/src/app/TelemetryService.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ApplicationInsights,
+  SeverityLevel,
+} from "@microsoft/applicationinsights-web";
+import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
+import { useHistory } from "react-router";
+
+let reactPlugin: ReactPlugin | null = null;
+let appInsights: ApplicationInsights | null = null;
+
+const createTelemetryService = () => {
+  const initialize = (browserHistory?: ReturnType<typeof useHistory>) => {
+    const instrumentationKey = process.env.REACT_APP_APPINSIGHTS_KEY;
+
+    if (!instrumentationKey) {
+      console.warn("Instrumentation key not provided");
+      return;
+    }
+
+    reactPlugin = new ReactPlugin();
+
+    appInsights = new ApplicationInsights({
+      config: {
+        instrumentationKey,
+        extensions: [reactPlugin],
+        loggingLevelConsole: process.env.NODE_ENV === "development" ? 2 : 0,
```

Comment:
I was able to confirm that setting `loggingLevelConsole` to 0 stops all console logging from AppInsights itself, which should prevent infinite loop concerns.

Reply:
Excellent! I just submitted a similar comment in my review; so we can just `resolve` that convo.

And with that, I say this whole situation is ✅